x-pack/plugins/siem/public/components/charts/areachart.tsx

```diff
@@ -0,0 +1,135 @@
+    ? {
+        area: {
+          fill: color,
+          opacity: 0.04,
+          visible: true,
+        },
+        line: {
+          stroke: color,
+          strokeWidth: 1,
+          visible: true,
+        },
+        border: {
+          visible: false,
+          strokeWidth: 1,
+          stroke: color,
+        },
+        point: {
+          visible: false,
+          radius: 0.2,
+          stroke: color,
+          strokeWidth: 1,
+          opacity: 1,
+        },
+      }
+    : undefined;
+};
+
+export const AreaChartBaseComponent = pure<{
+  data: AreaChartData[];
+  width: number | null | undefined;
+  height: number | null | undefined;
+}>(({ data, ...chartConfigs }) => {
+  return chartConfigs.width && chartConfigs.height ? (
+    <div style={{ height: chartConfigs.height, width: chartConfigs.width, position: 'relative' }}>
+      <Chart>
+        {data.map((series, idx) => {
+          const seriesKey = series.key;
+          const seriesSpecId = getSpecId(seriesKey);
+          return series.value != null ? (
+            <AreaSeries
+              id={seriesSpecId}
+              key={seriesKey}
+              name={series.key.replace('Histogram', '')}
+              data={series.value}
+              xScaleType={ScaleType.Ordinal}
+              yScaleType={ScaleType.Linear}
+              xAccessor="x"
+              yAccessors={['y']}
+              areaSeriesStyle={getSeriesLineStyle(series.color)}
+              customSeriesColors={getSeriesStyle(seriesKey, series.color)}
+            />
+          ) : null;
+        })}
+
+        <Axis
+          id={getAxisId(`group-${data[0].key}-x`)}
+          position={Position.Bottom}
+          showOverlappingTicks={false}
+          tickFormat={dateFormatter}
+          tickSize={0}
+        />
+        <Axis
+          id={getAxisId(`group-${data[0].key}-y`)}
+          position={Position.Left}
+          tickSize={0}
+          tickFormat={numberFormatter}
+        />
+      </Chart>
+    </div>
+  ) : null;
+});
+
+export const AreaChartWithCustomPrompt = pure<{
+  data: AreaChartData[] | null | undefined;
+  height: number | null | undefined;
+  width: number | null | undefined;
+}>(({ data, height, width }) => {
+  return data != null &&
+    data.length &&
+    data.every(
+      ({ value }) =>
+        value != null &&
+        value.length > 0 &&
+        value.every(chart => chart.x != null && chart.y != null)
+    ) ? (
+    <AreaChartBaseComponent height={height} width={width} data={data} />
+  ) : (
+    <ChartHolder />
+  );
+});
+
+export const AreaChart = pure<{ areaChart: AreaChartData[] | null | undefined }>(
+  ({ areaChart }) => (
+    <AutoSizer detectAnyWindowResize={false} content>
+      {({ measureRef, content: { height, width } }) => (
+        <WrappedByAutoSizer innerRef={measureRef}>
+          <AreaChartWithCustomPrompt data={areaChart} height={height} width={width} />
+        </WrappedByAutoSizer>
+      )}
+    </AutoSizer>
+  )
+);
```

Comment:
Really clean and great code! For these `pure` functions I would recommend changing them to use `React.Memo` like so:

```ts
export const AreaChart = React.memo<{ areaChart: AreaChartData[] | null | undefined }>(
  ({ areaChart }) => (
    <AutoSizer detectAnyWindowResize={false} content>
      {({ measureRef, content: { height, width } }) => (
        <WrappedByAutoSizer innerRef={measureRef}>
          <AreaChartWithCustomPrompt data={areaChart} height={height} width={width} />
        </WrappedByAutoSizer>
      )}
    </AutoSizer>
  )
);
```

For new code moving forward I think we can remove our usage of recompose and just utilize React when we write new code which is pretty awesome.

Reply:
Sure thing, thanks! I am happy to move away from pure component. I'll have another PR for enhancing kpi network, and do the cleanup for it as well.